src/checkout-buttons/strategies/paypal-button-strategy.ts

```diff
@@ -0,0 +1,142 @@
+                    env,
+                    client: clientToken,
+                    commit: paypalOptions.shouldProcessPayment,
+                    funding: {
+                        allowed: allowedSources,
+                        disallowed: disallowedSources,
+                    },
+                    style: {
+                        shape: 'rect',
+                        ...pick(paypalOptions.style, 'layout', 'size', 'color', 'label', 'shape', 'tagline', 'fundingicons'),
+                    },
+                    payment: (data, actions) => this._setupPayment(paymentMethod.config.merchantId, actions, paypalOptions.onPaymentError),
+                    onAuthorize: (data, actions) => this._tokenizePayment(data, actions, paypalOptions.shouldProcessPayment, paypalOptions.onAuthorizeError),
+                }, options.containerId);
+            })
+            .then(() => super.initialize(options));
+    }
+
+    deinitialize(): Promise<void> {
+        if (!Object.keys(this._isInitialized).length) {
+            return super.deinitialize();
+        }
+
+        this._paymentMethod = undefined;
+
+        return super.deinitialize();
+    }
+
+    private _setupPayment(merchantId: string | undefined, actions?: PaypalActions, onError?: (error: StandardError) => void): Promise<string> {
+        return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
+            .then(state => {
+                const checkout = state.checkout.getCheckout();
+                const config = state.config.getStoreConfig();
+
+                if (!actions) {
+                    throw new NotInitializedError(NotInitializedErrorType.CheckoutButtonNotInitialized);
+                }
+
+                if (!checkout) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCheckout);
+                }
+
+                if (!config) {
+                    throw new MissingDataError(MissingDataErrorType.MissingCheckoutConfig);
+                }
+
+                return actions.request.post('/api/storefront/paypal-payment/', { merchantId });
+            })
+            .then(res => res.id)
+            .catch(error => {
+                if (onError) {
+                    onError(error);
+                }
+
+                throw error;
+            });
+    }
+
+    private _tokenizePayment(
+        data: PaypalAuthorizeData,
+        actions?: PaypalActions,
+        shouldProcessPayment?: boolean,
+        onError?: (error: StandardError) => void
+    ): Promise<void> {
+        if (!this._paymentMethod) {
+            throw new NotInitializedError(NotInitializedErrorType.CheckoutButtonNotInitialized);
+        }
+
+        if (!actions) {
+            throw new NotInitializedError(NotInitializedErrorType.CheckoutButtonNotInitialized);
+        }
+
+        const methodId = this._paymentMethod.id;
+
+        return actions.payment.get(data.paymentID || '').then(payload => {
```

Comment:
Does it work if you call `actions.payment.get` without a payment ID? If not, is it better if you don't make that request at all (i.e.: do a null check first)?